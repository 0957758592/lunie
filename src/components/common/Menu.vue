<template>
  <div>
    <DesktopMenu id="desktop-menu" :links="links" />
    <MobileMenu id="mobile-footer-menu" :links="links" />
  </div>
</template>

<script>
import DesktopMenu from "common/DesktopMenu"
import MobileMenu from "common/MobileMenu"

export default {
  name: `main-menu`,
  components: {
    DesktopMenu,
    MobileMenu
  },
  data: () => ({
    links: [
      {
        icon: "account_balance_wallet",
        route: "/staking/my-delegations",
        title: "Portfolio"
      },
      {
        icon: "add_box",
        route: "/staking/validators",
        title: "Validators"
      },
      {
        icon: "swap_horiz",
        route: "/transactions",
        title: "Activity"
      },
      {
        icon: "how_to_vote",
        route: "/governance/proposals",
        title: "Proposals"
      }
    ]
  })
}
</script>

<style scoped></style>
