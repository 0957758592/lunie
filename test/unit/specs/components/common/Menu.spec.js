import { shallowMount } from "@vue/test-utils"
import Menu from "common/Menu"

describe(`Menu`, () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(Menu)
  })

  it("should show a the menus for mobile and desktop", () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
