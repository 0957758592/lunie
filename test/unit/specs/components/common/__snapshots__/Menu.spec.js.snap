// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should show a the menus for mobile and desktop 1`] = `
<div>
  <desktopmenu-stub
    id="desktop-menu"
    links="[object Object],[object Object],[object Object],[object Object]"
  />
   
  <mobilemenu-stub
    id="mobile-footer-menu"
    links="[object Object],[object Object],[object Object],[object Object]"
  />
</div>
`;
